tests/integration/ha/test_backups.py

```diff
@@ -0,0 +1,568 @@
+    # Deploy TLS Certificates operator.
+    config = {"ca-common-name": "CN_CA"}
+    await ops_test.model.deploy(TLS_CERTIFICATES_APP_NAME, channel="stable", config=config),
+
+    s3_charm = S3_INTEGRATOR
+    # Convert to integer as environ always returns string
+    app_num_units = 3
+    await asyncio.gather(
+        ops_test.model.deploy(s3_charm, channel=S3_INTEGRATOR_CHANNEL),
+        ops_test.model.deploy(my_charm, num_units=app_num_units, series=SERIES),
+    )
+
+    # Relate it to OpenSearch to set up TLS.
+    await ops_test.model.integrate(APP_NAME, TLS_CERTIFICATES_APP_NAME)
+    await ops_test.model.wait_for_idle(
+        apps=[TLS_CERTIFICATES_APP_NAME, APP_NAME],
+        status="active",
+        timeout=1400,
+        idle_period=IDLE_PERIOD,
+    )
+    # Credentials not set yet, this will move the opensearch to blocked state
+    # Credentials are set per test scenario
+    await ops_test.model.integrate(APP_NAME, S3_INTEGRATOR)
+
+
+@pytest.mark.group("all")
+@pytest.mark.abort_on_fail
+async def test_repo_missing_message(ops_test: OpsTest) -> None:
+    """Check the repo is missing."""
+    unit_ip = await get_leader_unit_ip(ops_test)
+    resp = await http_request(
+        ops_test, "GET", f"https://{unit_ip}:9200/_snapshot/{S3_REPO_NAME}", json_resp=True
+    )
+    logger.info(f"Response: {resp}")
+    assert resp["status"] == 404
+    assert "repository_missing_exception" in resp["error"]["type"]
+
+
+@pytest.mark.group("all")
+@pytest.mark.abort_on_fail
+async def test_repo_is_misconfigured(ops_test: OpsTest) -> None:
+    """Check the repo is misconfigured."""
+    unit_ip = await get_leader_unit_ip(ops_test)
+    app = (await app_name(ops_test)) or APP_NAME
+
+    config = {
+        "endpoint": "http://localhost",
+        "bucket": "error",
+        "path": "/",
+        "region": "default",
+    }
+    credentials = {
+        "access-key": "error",
+        "secret-key": "error",
+    }
+
+    # Not using _configure_s3 as this method will cause opensearch to block
+    await ops_test.model.applications[S3_INTEGRATOR].set_config(config)
+    await run_action(
+        ops_test,
+        0,
+        "sync-s3-credentials",
+        params=credentials,
+        app=S3_INTEGRATOR,
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[S3_INTEGRATOR],
+        status="active",
+        timeout=TIMEOUT,
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[app],
+        timeout=TIMEOUT,
+    )
+
+    resp = await http_request(
+        ops_test, "GET", f"https://{unit_ip}:9200/_snapshot/{S3_REPO_NAME}/_all", json_resp=True
+    )
+    logger.info(f"Response: {resp}")
+    assert resp["status"] == 500
+    assert "repository_exception" in resp["error"]["type"]
+    assert "Could not determine repository generation from root blobs" in resp["error"]["reason"]
+
+
+@pytest.mark.group("all")
+@pytest.mark.abort_on_fail
+async def test_backup_and_switch_configs(
```

Comment:
i'm not entirely sure I understand what we're testing here?

Reply:
@Mehdi-Bendriss this tests config changed, as: I change `s3-integrator` to point to: microceph, then aws and an incorrect config; and see how it behaves